app/org/catalyst/slackservice/controllers/EventController.java

```diff
@@ -248,4 +215,117 @@ private static boolean isInvalidUrlVerification(final Request request) {
     private static boolean isInvalidEventCallback(final Request request){
         return request.type.equals(TYPE_EVENT_CALLBACK) && (request.event == null || (!SUBTYPE_TOKENS_REVOKED.equals(request.event.type) && request.event.channel == null));
     }
+
+    private enum RequestHandler {
+        InvalidRequest { /* Not used in practice */
+            @Override
+            public CompletionStage<Result> handle(EventController controller, Http.Request httpRequest, MessageHandler messages, Request request, Logger logger) {
+                return ResultHelper.badRequest(messages, MessageHandler.INVALID_REQUEST);
+            }
+
+            @Override
+            public boolean matches(Request request, Http.Request httpRequest, AppConfig config) {
+                return false;
+            }
+        },
```

Comment:
Would it be worth replacing the `ResultHelper.badRequest(messages, MessageHandler.INVALID_REQUEST);` line above with `RequestHandler.InvalidRequest.handle(controller, httpRequest, messages, new Request(), logger)`? 
-or-
Potentially overload handle to accept an exception and return an invalid request?